### PR TITLE
feat(argo-cd): Use new Argo CD extension mechanism

### PR DESCRIPTION
**Pull request by yu-croco**
Resolves https://github.com/argoproj/argo-helm/issues/2396 

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/templates/argocd-server/deployment.yaml [hunk 1]
```diff
@@ -357,13 +357,22 @@ spec:
         lifecycle:
           {{- toYaml . | nindent 10 }}
         {{- end }}
+      {{- with .Values.server.extraContainers }}
+        {{- tpl (toYaml .) $ | nindent 6 }}
+      {{- end }}
+      {{- if or .Values.server.initContainers (and .Values.server.extensions.enabled .Values.server.extensions.extensionList) }}
+      initContainers:
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:364]: It's so naive 🙃

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -1574,29 +1574,41 @@ server:
...
+    # -- Extensions for Argo CD
+    # @default -- `[]` (See [values.yaml])
+    ## Ref: https://github.com/argoproj-labs/argocd-extension-metrics#install-ui-extension
+    extensionList: []
+    #  - name: extension-metrics
+    #    env:
+    #      - name: EXTENSION_URL
+    #        value: https://github.com/argoproj-labs/argocd-extension-metrics/releases/download/v1.0.0/extension.tar.gz
+    #      - name: EXTENSION_CHECKSUM_URL
+    #        value: https://github.com/argoproj-labs/argocd-extension-metrics/releases/download/v1.0.0/extension_checksums.txt
```
Comment (yu-croco) [inline charts/argo-cd/values.yaml:1603]: Original sample -> https://github.com/argoproj-labs/argocd-extension-metrics?tab=readme-ov-file#install-ui-extension

File: charts/argo-cd/values.yaml [hunk 3]
```diff
@@ -1574,29 +1574,41 @@ server:
...
+
     # -- Server UI extensions container-level security context
     # @default -- See [values.yaml]
     containerSecurityContext:
       runAsNonRoot: true
       readOnlyRootFilesystem: true
       allowPrivilegeEscalation: false
+      runAsUser: 1000
```
Comment (yu-croco) [inline charts/argo-cd/values.yaml:1611]: I got `container has runAsNonRoot and image has non-numeric user ...` , so I added as following [sample](https://github.com/argoproj-labs/argocd-extension-metrics?tab=readme-ov-file#install-ui-extension)

File: charts/argo-cd/templates/aggregate-roles.yaml [hunk 4]
```diff
@@ -14,9 +14,6 @@ rules:
   {{- if .Values.applicationSet.enabled }}
   - applicationsets
   {{- end }}
-  {{- if .Values.server.extensions.enabled }}
-  - argocdextensions
-  {{- end }}
```
Comment (yu-croco) [inline charts/argo-cd/templates/aggregate-roles.yaml:19]: argocdextensions  CRD is removed ♻️

**Review by tico24 — APPROVED** [review]

**Review by pdrastil — APPROVED** [review]

**Review by jmeridth — APPROVED** [review]

**Comment by mkilchhofer** [comment]
👍🚀

File: charts/argo-cd/templates/aggregate-roles.yaml (hunk 4 above)
Comment (nerzhul) [reply charts/argo-cd/templates/aggregate-roles.yaml:19]: to be honest, it should be a breaking release of the chart

Comment (yu-croco) [reply charts/argo-cd/templates/aggregate-roles.yaml:19]: Yes, so I bumped minor version as following [Versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning) and put changelog in README.
Do you want us to do more things ?

**Pull request merged**